component/src/main/java/org/ballerinalang/kafka/util/KafkaUtils.java

```diff
@@ -518,16 +520,32 @@ private static void addBooleanParamIfPresent(String paramName,
 
     public static List<BMap<String, BValue>> createPartitionList(Context context,
                                                                  Collection<TopicPartition> partitions) {
+
         List<BMap<String, BValue>> topicPartitionList = new ArrayList<>();
         if (!partitions.isEmpty()) {
             partitions.forEach(assignment -> {
-                BMap<String, BValue> partitionStruct = KafkaUtils.
-                        createKafkaPackageStruct(context, TOPIC_PARTITION_STRUCT_NAME);
-                partitionStruct.put("topic", new BString(assignment.topic()));
-                partitionStruct.put("partition", new BInteger(assignment.partition()));
+                BMap<String, BValue> partitionStruct = getTopicPartitionStruct(context, assignment);
                 topicPartitionList.add(partitionStruct);
             });
         }
         return topicPartitionList;
     }
+
+    public static BMap<String, BValue> getTopicPartitionStruct(Context context, TopicPartition topicPartition) {
+        BMap<String, BValue> partitionStruct = KafkaUtils
+                .createKafkaPackageStruct(context, TOPIC_PARTITION_STRUCT_NAME);
+        partitionStruct.put("topic", new BString(topicPartition.topic()));
```

Comment:
Shouldn't we need to have constants for topic and the partition?

Reply:
Agreed. Changes done.